server-phoenix/test/helios_web/controllers/web_hooks/slack_controller_test.exs

```diff
@@ -105,6 +108,172 @@ defmodule HeliosWeb.WebHooks.SlackControllerTest do
     assert Map.fetch(first_message, :source_author) == {:ok, "Priscila"}
   end
 
+  test "channel name (not nil) was added to db with id", %{conn: conn} do
+    body = %{
+      "channel" => "asdf"
+    }
+
+    request_body = URI.encode_query(body)
+
+    headers = [
+      {"Accept", "application/json"},
+      {"Authorization", "Bearer asdf"},
+      {"Content-Type", "application/x-www-form-urlencoded"}
+    ]
+
+    with_mock HTTPoison,
+      post!: fn "https://slack.com/api/conversations.info",
+                request_body,
+                headers,
+                follow_redirect: true ->
+        %HTTPoison.Response{
+          body: "{
+            \"ok\": \"true\",
+            \"channel\": {
+                \"name\": \"general\"
+            }
+        }"
+        }
+      end do
+      conn
+      |> put_req_header("content-type", "application/json")
+      |> post(
+        Routes.slack_path(conn, :handle),
```

Comment:
If you move `insert_channel_id` into its own module, you can test that in isolation rather than posting the controller